### PR TITLE
maint: update otel dependencies to 1.25.0

### PR DESCRIPTION
**Pull request by codeboten**
## Which problem is this PR solving?

Update OTel Go dependencies to 1.25.0

## Short description of the changes

This change has the side effect of bumping the minimum version to go 1.21.0

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-go 1.20
+go 1.21
 
 require (
 	github.com/honeycombio/otel-config-go v1.13.1
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/otel v1.24.0
-	go.opentelemetry.io/otel/exporters/stdout/stdouttrace v1.23.0
-	go.opentelemetry.io/otel/sdk v1.23.0
+	go.opentelemetry.io/otel v1.25.0
```
Comment (JamieDanielson) [inline go.mod:8]: Should we wait and update otel-config-go once it's released?

File: README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Latest release built with:
 - [OpenTelemetry v1.23/v0.48.0](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.23.0)
 - [OTel Config v1.13.1](https://github.com/honeycombio/otel-config-go/releases/tag/v1.13.1)
 
-Minimum Go Version: `1.20`
+Minimum Go Version: `1.21`
```
Comment (JamieDanielson) [inline README.md:17]: We typically hold off on changing this until it's time to release, as folks sometimes use the readme for reference with whatever is the latest release.
```suggestion
Minimum Go Version: `1.20`
```

File: go.mod (hunk 1 above)
Comment (MikeGoldsmith) [reply go.mod:8]: I think it's fine to let this change happen and update otel-config-go in a separate PR. The dependency resolver will figure out the min otel version between direct & indirect deps.

**Review by MikeGoldsmith — APPROVED** [review]
Looks good to me. I agree with @JamieDanielson we should update otel-config-go before releasing a new version of this distro to ensure deps are aligned.

File: go.mod (hunk 1 above)
Comment (codeboten) [reply go.mod:8]: Updated otel-config-go as well

**Pull request merged**